### ISSUE TITLE
Swagger-ui does not hide readOnly nested object from example body

### ISSUE THREAD

**Issue by Yaroslav007** [swagger-api/swagger-ui#5418]
I'm developing spring-boot app with swagger-ui and lombok. Currently I'm trying to hide nested object from request body, but it still shows in example json on swagger-ui page.

I have my class with annotations(simplified to only related stuff):

```
@Setter
@Getter
@ApiModel(description = "Character model")
public class Character {

    @ApiModelProperty(readOnly = true)
    private Long id;

    @ApiModelProperty(readOnly = true)
    private SearchAnnouncement searchAnnouncement;
}
```

When I access example model on swagger page "id" filed is properly hidden in example json, and visible in response model. But "searchAnnouncement" is not hidden in example json.

I tried:
using readOnly = true
using hidden = true
using accessMode = ApiModelProperty.AccessMode.READ_ONLY
using @Setter(AccessLevel.NONE), which prevents generation of setter for that field, as I read somewhere that something is checking if setter exist and sets readOnly based on that
different combinations of above
but in all cases example json looks like this (id hidden, and searchAnnouncement visible):

```
{
    "searchAnnouncement": {
         "id": 0,
    },
}
```
So in short my question is:
Is it possible to hide from example json nested object?

### Q&A (please complete the following information)
 - OS: [e.g. windows]
 - Browser: [Chrome ]
 - Version: [74]
 - Swagger-UI version: [2.9.2]

**Comment by PKapski** [comment]
I am having the same problem, is there any progress?

**Comment by anandvarkeyphilips** [comment]
@Yaroslav007 , found any work around or is it still happening?

**Comment by Yaroslav007** [comment]
> @Yaroslav007 , found any work around or is it still happening?

I didn't find a workaround for it

**Comment by akuma8** [comment]
Any solution?

**Comment by NadavK** [comment]
See workaround here: https://stackoverflow.com/questions/51402156/how-to-declare-a-ref-property-as-readonly-in-openapi-swagger

**Comment by ospikovets** [comment]
Is there any way to remove `readOnly` property from the UI _Request body_ -> _Schema_ section?

Even though `id` has `readOnly` notation, it is still weird to have it in the request body schema.
![image](https://user-images.githubusercontent.com/10669216/81425422-404e9080-9160-11ea-89f6-f4d4983c7843.png)

**Comment by cnizzardini** [comment]
This would be really nice to resolve, my current workaround is just to include the properties manually. But this creates REALLY redundant output.

**Comment by shwetatyagi22** [comment]
I am generating models using openapitools. is there any way to hide readonly nested objects in swagger ui.

**Comment by flmelody** [comment]
Any solution now?

**Comment by cnizzardini** [comment]
I eventually get around this by creating multiple schemas per model: read, write, add and edit. These were hidden from swagger by placing them in a vendor location like #/x-my-application/components/schemas.

hidden schemas:

```yaml
      Actor-Write:
        properties:
          first_name:
            minLength: 1
            maxLength: 64
            description: Actor First Name
            type: string
          last_name:
            minLength: 1
            maxLength: 64
            description: Actor Last Name
            type: string
        description: Actor Entity
        type: object
      Actor-Add:
        required:
          - first_name
          - last_name
        allOf:
          - $ref: '#/x-swagger-bake/components/schemas/Actor-Write'
        description: Actor Entity
        type: object
      Actor-Edit:
        allOf:
          - $ref: '#/x-swagger-bake/components/schemas/Actor-Write'
        description: Actor Entity
        type: object
```

regular schema:

```yaml
    Actor:
      properties:
        id:
          readOnly: true
          type: integer
          format: int64
        first_name:
          minLength: 1
          maxLength: 64
          description: Actor First Name
          type: string
        last_name:
          minLength: 1
          maxLength: 64
          description: Actor Last Name
          type: string
        modified:
          readOnly: true
          minLength: 1
          description: Last modified date/time
          type: string
          format: date-time
      description: Actor Entity
      type: object
```

request body:

```yaml
      requestBody:
        content:
          application/x-www-form-urlencoded:
            schema:
              $ref: '#/x-swagger-bake/components/schemas/FilmActor-Add'
          application/json:
            schema:
              $ref: '#/x-swagger-bake/components/schemas/FilmActor-Add'
          application/xml:
            schema:
              $ref: '#/x-swagger-bake/components/schemas/FilmActor-Add'
```

Obviously, this is a bit of work to do manual, but I built this into a library: https://github.com/cnizzardini/cakephp-swagger-bake and a demo is available here: https://demo.mixerapi.com/admin

**Comment by flmelody** [comment]
> I eventually got around this by creating multiple schemas per model: read, write, add and edit. These were hidden from swagger by placing them in a vendor location like #/x-my-application/components/schemas.
> 
> hidden schemas:
> 
> ```yaml
>       Actor-Write:
>         properties:
>           first_name:
>             minLength: 1
>             maxLength: 64
>             description: Actor First Name
>             type: string
>           last_name:
>             minLength: 1
>             maxLength: 64
>             description: Actor Last Name
>             type: string
>         description: Actor Entity
>         type: object
>       Actor-Add:
>         required:
>           - first_name
>           - last_name
>         allOf:
>           - $ref: '#/x-swagger-bake/components/schemas/Actor-Write'
>         description: Actor Entity
>         type: object
>       Actor-Edit:
>         allOf:
>           - $ref: '#/x-swagger-bake/components/schemas/Actor-Write'
>         description: Actor Entity
>         type: object
> ```
> 
> regular schema:
> 
> ```yaml
>     Actor:
>       properties:
>         id:
>           readOnly: true
>           type: integer
>           format: int64
>         first_name:
>           minLength: 1
>           maxLength: 64
>           description: Actor First Name
>           type: string
>         last_name:
>           minLength: 1
>           maxLength: 64
>           description: Actor Last Name
>           type: string
>         modified:
>           readOnly: true
>           minLength: 1
>           description: Last modified date/time
>           type: string
>           format: date-time
>       description: Actor Entity
>       type: object
> ```
> 
> request body:
> 
> ```yaml
>       requestBody:
>         content:
>           application/x-www-form-urlencoded:
>             schema:
>               $ref: '#/x-swagger-bake/components/schemas/FilmActor-Add'
>           application/json:
>             schema:
>               $ref: '#/x-swagger-bake/components/schemas/FilmActor-Add'
>           application/xml:
>             schema:
>               $ref: '#/x-swagger-bake/components/schemas/FilmActor-Add'
> ```
> 
> This is a bit of work to do manually so I built this into a library that does the work for me: https://github.com/cnizzardini/cakephp-swagger-bake and a demo is available here: https://demo.mixerapi.com/admin

thanks for reply,I may create another model for it current time,your theme color is beautiful!
